### PR TITLE
Add verification badges to `Hub`

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds block verification badges to the hub page and to block pages.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203179076056209/1203623307402534/f) _(internal)_
- [Asana task](https://app.asana.com/0/0/1203966325298247/f) _(internal)_

## 🔍 What does this change?

- Added `verified: true` to local blocks;
- Display the badge if `verified` is set to `true` in the hub and block page.

## ❓ How to test this?

To test local blocks:
1.  Run `yarn workspace @apps/site exe scripts/prepare-blocks.ts`;
2. Check if blocks in the hub are displaying the badge.

To test published blocks:
1. Publish a random block (message me via slack if you need help with doing this locally);
2. Add a `verified: true` to it directly in MongoDB;
3. Check if blocks in the hub are displaying the badge.

## 📹 Demo

image

image

### CODE REVIEW

**Comment by nathggns** [comment]
Reviewing

File: apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx [hunk 1]
```diff
@@ -301,10 +302,46 @@ const BlockPage: NextPage<BlockPageProps> = ({
...
+                <Box component="span">{blockMetadata.displayName} </Box>
+
+                {blockMetadata.verified ? (
+                  <Box
+                    sx={{
+                      display: "inline-flex",
+                      alignItems: "center",
+                      gap: 0.5,
+                      borderRadius: 7.5,
+                      background: "#FBF7FF",
```
Comment (nathggns) [inline apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx:323]: We should look at naming this colour at some point as its being used more and more

**Review by nathggns — APPROVED** [review]

**Review by kachkaev — APPROVED** [review]

**Review by kachkaev — CHANGES_REQUESTED** [review]
Hmm do we have a mechanism to get the `verified` field for blocks that are published from the API? `/hub` directory will be removed in https://github.com/blockprotocol/blockprotocol/pull/936.

File: apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx [hunk 2]
```diff
@@ -301,10 +302,46 @@ const BlockPage: NextPage<BlockPageProps> = ({
...
+                      variant="bpMicroCopy"
+                      sx={{
+                        fontWeight: 600,
+                        fontSize: 13,
+                        lineHeight: "18px",
+                        textTransform: "uppercase",
+                        color: "#7A4FF5",
+                      }}
+                    >
+                      Verified
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx:338]: I agree. @nonparibus what do you think?

**Comment by luisbettencourt** [comment]
> Hmm do we have a mechanism to get the `verified` field for blocks that are published from the API? `/hub` directory will be removed in #936.

I see. Are those jsons being moved, or outright removed? If they are getting removed where will that information come from?

File: apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nonparibus) [reply apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx:338]: Agreed. Please ticketize for follow-up. No need to block on this.

**Comment by nonparibus** [comment]
@luisbettencourt Blocks won't be able to self-declare they are verified. This is information that will be added to a database record per relevant Hub listing. cc @CiaranMn

**Comment by luisbettencourt** [comment]
> @luisbettencourt Blocks won't be able to self-declare they are verified. This is information that will be added to a database record per relevant Hub listing. cc @CiaranMn

@CiaranMn, given this should I remove the code that handles looking for the `verified` field in the block's json, or do we not care that it stays?

File: apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx:338]: Created a [subtask ](https://app.asana.com/0/0/1203966325298247/f) to track this.

[the remark this answers is not in the document]
Comment (luisbettencourt) [reply apps/site/src/pages/[shortname]/blocks/[block-slug].page.tsx]: @nonparibus, went ahead and added the tooltip in this PR, as it was a quick addition to make. Just let me know what the Tooltip should read as, right now it reads: `This block is verified`

**Comment by CiaranMn** [comment]
> @CiaranMn, given this should I remove the code that handles looking for the `verified` field in the block's json, or do we not care that it stays?

I don't think it matters either way, they'll be gone in a few days anyway.

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**